### DISCUSSION TITLE
How to target a specific field using Property's onUpdate?

### DISCUSSION THREAD

**Discussion by Alwinator** [mikro-orm/mikro-orm#3123] [Q&A] [answered]
My entity:
```ts
@Property()
name: string

@Property()
password: string

/* Only run onUpdate when the password field changes */
@Property({ type: DateTimeType, onUpdate: () => new Date() })
lastPasswordChangeAt: Date = new Date()
```

I only want to update lastPasswordChangeAt when the password field changes. When other fields such as name change, the date should stay the same. How can I implement this behavior using MikroORM?

**Comment by B4nan** [comment]
Use hooks instead, there you get the full changeset object as one of the parameters.

**Comment by Alwinator** [comment, marked as answer]
@B4nan Thank you for the hint with the hooks. I opted for the following solution:
```ts
@BeforeCreate()
@BeforeUpdate()
async updatePassword(event) {
    const { payload } = event.changeSet

    if(payload.password){
      this.lastPasswordChangeAt = new Date()
    }
}
```
